core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.spec.ts

```diff
@@ -1,22 +1,267 @@
+                    }
+                },
+                {
+                    provide: SiteService,
+                    useValue: {
+                        get currentSite() {
+                            return undefined;
+                        },
 
+                        get switchSite$() {
+                            return switchSiteSubject.asObservable();
+                        }
+                    }
+                },
+                StringUtils,
+                DotHttpErrorManagerService,
+                DotAlertConfirmService,
+                ConfirmationService,
+                LoginService,
+                DotcmsEventsService,
+                DotEventsSocket,
+                DotcmsConfigService,
+                DotMessageDisplayService,
+                DialogService,
+                DotSiteBrowserService,
+                { provide: DotFormatDateService, useClass: DotFormatDateServiceMock }
+            ],
+            imports: [
+                DotListingDataTableModule,
+                CommonModule,
+                DotMessagePipeModule,
+                SharedModule,
+                CheckboxModule,
+                MenuModule,
+                ButtonModule,
+                DotActionButtonModule,
+                DotActionMenuButtonModule,
+                DotAddToBundleModule,
+                HttpClientTestingModule,
+                DynamicDialogModule,
+                BrowserAnimationsModule
+            ],
+            schemas: [CUSTOM_ELEMENTS_SCHEMA]
+        }).compileComponents();
         fixture = TestBed.createComponent(ContainerListComponent);
-        component = fixture.componentInstance;
+        dotPushPublishDialogService = TestBed.inject(DotPushPublishDialogService);
+        dialogService = TestBed.inject(DialogService);
+        coreWebService = TestBed.inject(CoreWebService);
+    });
+
+    it('should reload portlet only when the site change', () => {
+        const checkbox = fixture.debugElement.query(
+            By.css('[data-testId="archiveCheckbox"]')
+        ).componentInstance;
+
         fixture.detectChanges();
+
+        expect(checkbox.binary).toBeTruthy();
     });
 
-    it('should create', () => {
-        expect(component).toBeTruthy();
+    describe('with data', () => {
+        beforeEach(fakeAsync(() => {
+            spyOn<any>(coreWebService, 'requestView').and.returnValue(
+                of({
+                    entity: containersMock,
+                    header: (type) => (type === 'Link' ? 'test;test=test' : '10')
+                })
+            );
+            fixture.detectChanges();
+            tick(2);
+            fixture.detectChanges();
+            dotListingDataTable = fixture.debugElement.query(
+                By.css('dot-listing-data-table')
+            ).componentInstance;
+            spyOn(dotPushPublishDialogService, 'open');
+
+            spyOn<any>(dialogService, 'open').and.returnValue({
```

Comment:
Maybe:

```suggestion
            spyOn<CoreWebService>(dialogService, 'open').and.returnValue({
```

Reply:
Not letting me use the type DialogService here, throws an error `Argument of type 'string' is not assignable to parameter of type 'never'.`